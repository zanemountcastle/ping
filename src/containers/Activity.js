import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Header } from 'react-native-elements';

export default class Activity extends Component {

  render() {
      return (
        <View style={styles.container}>
          <Header
            centerComponent={{ text: 'Activity', style: { color: '#4D4C4C', fontSize: 20, fontWeight: '700' } }}
            outerContainerStyles={{ backgroundColor: '#fff' }}
          />
          <View style={styles.body}>
            <Text>Activity page!</Text>
          </View>
        </View>
      );
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    flexDirection: 'column',
  },
  body: {
    flex: 1,
    backgroundColor: '#fff',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
