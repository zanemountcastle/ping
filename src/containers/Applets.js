import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Header } from 'react-native-elements';

import AppletsFeed from '../components/AppletsFeed';

export default class Applets extends Component {

  render() {
    return (
      <View style={styles.container}>
        <Header
          leftComponent={{ icon: 'settings', color: '#4D4C4C' }}
          centerComponent={{ text: 'My Applets', style: { color: '#4D4C4C', fontSize: 20, fontWeight: '700' } }}
          rightComponent={{ icon: 'add', color: '#4D4C4C' }}
          outerContainerStyles={{ backgroundColor: '#fff' }}
        />
        <AppletsFeed />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
