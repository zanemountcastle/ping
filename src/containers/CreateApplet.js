import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Button, Header, FormLabel, FormInput } from 'react-native-elements';
import { uuid, randomColor } from '../lib/Utilities';

// Redux
import { connect } from 'react-redux';
import { CreateAnApplet } from '../actions';

export class CreateApplet extends Component {
  constructor(props) {
    super(props);
    this.state = {
        appletId: '',
        color: randomColor(),
        message: '',
        organization: 'NYUAD'
    };
  }

  _onFormSubmission() {
      const s = this.state;
      const authKey = uuid();
      this.props.CreateAnApplet(
        s.appletId, s.message, s.color, s.organization, authKey);
  }

  render() {
      return (
        <View style={styles.container}>
          <Header
            centerComponent={{ text: 'Create an Applet', style: { color: '#4D4C4C', fontSize: 20, fontWeight: '700' } }}
            outerContainerStyles={{ backgroundColor: '#fff' }}
          />
          <FormLabel>Message</FormLabel>
          <FormInput onChangeText={( m) => {this.setState({message: m}) }}/>

          <FormLabel>Organization</FormLabel>
          <FormInput
            defaultValue={this.state.organization}
            onChangeText={(o) => {this.setState({organization: o}) }}
          />

          <FormLabel>Applet ID</FormLabel>
          <FormInput onChangeText={(a) => {this.setState({appletId: a}) }}/>

          <FormLabel>Color</FormLabel>
          <FormInput
            defaultValue={this.state.color}
            onChangeText={(c) => {this.setState({color: c}) }}
          />

          <Button
            onPress={() => {this._onFormSubmission()}}
            title='Create applet'
            buttonStyle={styles.button}
            color={'#FFF'}
            loading={this.props.create.isFetching}
            disabled={this.props.create.isFetching}
          />

        </View>
      );
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    flexDirection: 'column',
  },
  body: {
    flex: 1,
    backgroundColor: '#fff',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    backgroundColor: '#FF7C7C',
    marginTop: 40,
    marginBottom: 5,
    alignSelf: 'center',
    width: 250,
    borderRadius: 5,
  },
});

function mapStateToProps(state) {
  return {
    create: state.create
  }
};

export default connect(mapStateToProps, { CreateAnApplet })(CreateApplet);
