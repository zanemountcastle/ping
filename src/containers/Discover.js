import React, { Component } from 'react';
import { StyleSheet, Text, View ,  ScrollView, Alert} from 'react-native';
import { Header } from 'react-native-elements';

import { connect } from 'react-redux';

import FetchDiscoveryData from '../actions/FetchDiscoveryData';

import { AppletsFeed, AppletPreview } from '../components';

export class Discover extends Component {
  componentWillMount() {
    this.props.FetchDiscoveryData();
  }

  render() {
      return (
        <View style={styles.container}>
          <Header
            centerComponent={{ text: 'Discover', style: { color: '#4D4C4C', fontSize: 20, fontWeight: '700' } }}
            outerContainerStyles={{ backgroundColor: '#fff' }}
          />
          <AppletsFeed
            feed={this.props.discovery.data}
            fetchApplets={this.props.FetchDiscoveryData}
            isFetching={this.props.discovery.isFetching}
          />
        </View>
      );
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    flexDirection: 'column',
  },
});

function mapStateToProps(state) {
  return {
    discovery: state.discovery
  }
};

export default connect(mapStateToProps, { FetchDiscoveryData })(Discover);
