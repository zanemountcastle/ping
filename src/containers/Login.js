import React from 'react';
import {
  ActivityIndicator,
  Image,
  Text,
  TextInput,
  StyleSheet,
  View} from 'react-native';
import * as firebase from 'firebase';
import {StackNavigator} from 'react-navigation';
import FCM from 'react-native-fcm';
import {Button, FormLabel, FormInput} from 'react-native-elements'

export default class login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "a@b.com",
      password: "12345678",
      error: null,
      loading: false
    };
    {
      console.ignoredYellowBox = ['Setting a timer']
    }
  }

  _onLoginPress() {
    this.setState({error: '', loading: true});

    const {email, password} = this.state;
    firebase.auth().signInWithEmailAndPassword(email, password).then(() => {
      this.setState({error: '', loading: false});
      FCM.getFCMToken().then(token => {
        userID = firebase.auth().currentUser.uid;
        firebase.database().ref('/users/' + userID).update({token: token });
      });
      this.props.navigation.navigate('Main');

    }).catch(() => {
      this.setState({error: 'Authentication failed', loading: false});
    });
  }

  _onSignUpPress() {
    this.props.navigation.navigate('Signup');
  }

  render() {
    return (
      <View style={styles.container}>
        <Image
          style={styles.logo}
          source={require('../../static/assets/logo.png')}
        />
        <View style={styles.form}>
          <Text style={styles.formHeader}>email</Text>
          <TextInput
            value={this.state.email}
            onChangeText={email => this.setState({email})}
            placeholder='john@icloud.com'
            style={styles.textInput}/>

          <Text style={styles.formHeader}>password</Text>
          <TextInput
            value={this.state.password}
            secureTextEntry
            placeholder='••••••••••'
            onChangeText={password => this.setState({password})}
            style={styles.textInput}
          />
          <Text>{this.state.error}</Text>
        </View>

        <Button
          onPress={this._onLoginPress.bind(this)}
          title='log in'
          buttonStyle={styles.login}
          color={'#000'}
          loading={this.state.loading}
          disabled={this.state.loading}
        />

        <Button
          onPress={this._onSignUpPress.bind(this)}
          title="sign up"
          color="#FFF"
          buttonStyle={styles.signup}
        />
      </View>
    )
  }

}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FF7C7C',
    flexDirection: 'column',
    justifyContent: 'center',
  },
  login: {
    backgroundColor: '#FFF',
    marginTop: 40,
    marginBottom: 5,
    alignSelf: 'center',
    width: 250,
    borderRadius: 5,
  },
  signup: {
    backgroundColor: 'transparent',
  },
  textInput: {
    width: 250,
    height: 120,
    borderRadius: 5,
    backgroundColor: '#FFF',
    alignSelf: 'center',
    height: 40,
    paddingLeft: 20,
  },
  formHeader: {
    marginTop: 20,
    color: '#FFF',
    fontWeight: '700',
    marginBottom: 2,
  },
  form: {
    alignSelf: 'center',
    width: 250,
  },
  logo: {
    width: 120,
    height: 120,
    alignSelf: 'center',
    marginBottom: 20,
  }
});
