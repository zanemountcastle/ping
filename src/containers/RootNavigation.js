// import { Notifications } from 'expo';
import React from 'react';
import { View, Button, Text } from 'react-native'
import { StackNavigator } from 'react-navigation';
import { LoginScreen } from '../components';
import { Tabs } from '../lib/Router';

const RootStackNavigator = StackNavigator(
  {
    Login:{
      screen: LoginScreen,

    },
    Main: {
      screen: Tabs,
    },

  },
  {
    headerMode: 'none',
  }
);

export default class RootNavigator extends React.Component {
  render() {
    return (
      <RootStackNavigator />
    );
  }
}
