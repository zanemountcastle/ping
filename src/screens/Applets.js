import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Header } from 'react-native-elements';

import AppletsFeed from '../components/AppletsFeed';
import AppletsHeader from '../components/AppletsHeader';

import { feeds } from '../config/data';

export default class Discover extends Component {

  render() {
    return (
      <View >
        <AppletsHeader />
        <AppletsFeed feeds={feeds} />
      </View>
    );
  }
}
