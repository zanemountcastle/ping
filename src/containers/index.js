import Activity from './Activity';
import Applets from './Applets';
import Discover from './Discover';
import Search from './Search';

export { Activity, Applets, Discover, Search };
