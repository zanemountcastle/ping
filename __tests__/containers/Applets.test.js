import React from 'react';

// Get named component, not default
import { Applets } from '../../src/containers/Applets';

import ShallowRenderer from 'react-test-renderer/shallow';

describe('my applets screen', () => {

  // Mock props that would be passed from Redux
  const appletsProps = {
    applets: {
      isFetching: false,
      data: [],
      errorMessage: null,
      hasError: false
    },
    FetchAppletData: () => {[]}
  }

  it('renders without crashing', () => {
    // Shallow render <Applets /> and not it's child components
    // Standard react-test-renderer doesn't allow shallow rendering
    const renderer = new ShallowRenderer();
    expect(renderer.render(<Applets {...appletsProps} />)).toBeTruthy();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    expect(renderer.render(<Applets {...appletsProps} />)).toMatchSnapshot();
  });

});
