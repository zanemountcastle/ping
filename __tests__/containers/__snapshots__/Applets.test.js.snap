// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my applets screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Header
    centerComponent={
      Object {
        "style": Object {
          "color": "#4D4C4C",
          "fontSize": 20,
          "fontWeight": "700",
        },
        "text": "My Applets",
      }
    }
    outerContainerStyles={
      Object {
        "backgroundColor": "#fff",
      }
    }
  />
  <AppletsFeed
    feed={Array []}
    fetchApplets={[Function]}
    isFetching={false}
  />
</View>
`;
